src/interface/fileList/fileList.js

```diff
@@ -2,21 +2,25 @@ const { getFileVisibility } = require('../../domain/file/utils');
 const File = require('../../domain/file');
 
 async function fileList(req, resp) {
-    const { invokerAddress } = req.query;
+    const { invokerAddress, ipfsHash } = req.query;
 
-    const files = await File.findAll(invokerAddress);
+    if (!invokerAddress && !ipfsHash) {
+        return resp.status(400).json({ error: 'invokerAddress or ipfsHash query parameter is required' });
+    }
+
+    const files = invokerAddress ? await File.findAll({ invokerAddress }) : await File.findAll({ ipfsHash });
```

Comment:
why are we using the findAll fn ?, are we expecting multiple metadata from a single ipfs hash

Reply:
We are using the findAll so as to keep the response structure same for both ipfsHash and invokerAddress.
If I call findOne, then will have to handle its response exclusively.